src/main/java/fi/helsinki/cs/tmc/core/TmcCore.java

```diff
@@ -147,41 +126,21 @@ public Path getExerciseChecksumCacheLocation() {
      * @param path where it downloads the exercises
      * @param courseId ID of course to download
      * @param observer ProgressObserver will be informed about the progress of downloading
-     * exercises. Observer can print progress status to end-user.
+     exercises. Observer can print progress status to end-user
      * @throws TmcCoreException if something in the given input was wrong
      */
     public ListenableFuture<List<Exercise>> downloadExercises(
             Path path, int courseId, ProgressObserver observer) throws TmcCoreException {
-        DownloadExercises downloadCommand =
-                new DownloadExercises(settings, path, courseId, observer, updateCache);
-        return threadPool.submit(downloadCommand);
-    }
 
-    /**
-     * Downloads exercise files specified in the given list. The exercises will be located in
-     * TmcMainDirectory (field in TmcSettings).
-     *
-     * @param exercises to be downloaded
-     */
-    public ListenableFuture<List<Exercise>> downloadExercises(List<Exercise> exercises)
-            throws TmcCoreException {
-        return this.downloadExercises(exercises, null);
-    }
-
-    /**
-     * Downloads exercises.
-     */
-    public ListenableFuture<List<Exercise>> downloadExercises(
```

Comment:
I'm unsure about whether we should remove this outright. A more cautious approach would be to `@deprecate` it first.